market_analysis/views/default.py

```diff
@@ -0,0 +1,450 @@
+                                    total_shares, round(total_value, 2)
+                                    )
+
+        export['stocks'] = stocks
+        export['total_shares'] = total_shares
+        export['total_value'] = round((total_value), 2)
+
+        return {'entry': export}
+
+    else:
+        return HTTPFound(request.route_url('api_error'))
+
+
+@view_config(route_name='new_user', renderer='templates/new_user.jinja2')
+def new_user(request):
+    """New User form."""
+    username = password = password_verify = first_name = ''
+    last_name = phone_number = email = error = message = ''
+
+    if request.method == 'POST':
+        username = request.POST['username']
+        password = request.POST['password']
+        password_verify = request.POST['password_verify']
+        first_name = request.POST['first_name']
+        last_name = request.POST['last_name']
+        phone_number = request.POST['phone_number']
+        email = request.POST['email']
+
+        try:
+            query = request.dbsession.query(Users)
+            result = query.filter_by(username=username).first()
+        except DBAPIError:  # pragma: no cover
+            return Response(db_err_msg, content_type='text/plain', status=500)
+
+        if result:
+            message = 'User "{}" already exists.'.format(username)
+        else:
+            if username != '' and password != '' and password_verify != ''\
+               and first_name != '' and last_name != '' and email != '':
+
+                if (password == password_verify) and (len(password) > 6):
+                    message = "good job, you can enter info"
+                    date_joined = datetime.datetime.now()
+                    date_last_logged = datetime.datetime.now()
+                    new = Users(
+                        username=username,
+                        first_name=first_name,
+                        last_name=last_name,
+                        email=email,
+                        email_verified=False,
+                        date_joined=date_joined,
+                        date_last_logged=date_last_logged,
+                        pass_hash=pwd_context.encrypt(password),
+                        phone_number=phone_number,
+                        phone_number_verified=False,
+                        active=True,
+                        password_last_changed=datetime.datetime.now(),
+                        password_expired=False,
+                        is_admin=0,
+                    )
+                    request.dbsession.add(new)
+                    headers = remember(request, username)
+                    return HTTPFound(location=request.route_url('search'),
+                                     headers=headers)
+                else:
+                    error = 'Passwords do not match or password'\
+                            ' is less than 6 characters'
+            else:
+                error = 'Missing Required Fields'
```

Comment:
I would much rather see these validations you pass through handled as sequential operations, rather than nested like this.  For example, consider a `validate_post` method:

```python
def validate_post_data(request):
    error = ''
    validated = {}
    missing_value_error = 'All fields are required'
    keys = ['username', 'password', 'password_verify', 'first_name', 'last_name', 'phone_number', 'email']
    try:
        validated = {key, request.POST[key] for key in keys}
    except KeyError:
        error = missing_value_error
    else:
        if not all(validated.values()):
            error = missing_value_error
            validated = {}
    return validated, error
```

You could test this with a number of different submitted data scenarios, and then it could be used like so:

```python
user_data, error = validate_post_data(request)
```
Then, add a `verify_passwords` function:

```python
def verify_passwords(user_data):
    bad_password_message = 'Passwords do not match, or password is less than 6 characters'
    if user_data['password'] != user_data['password_verify'] or len(user_data['password']) < 6:
        return bad_password_message
```
Now, you can go ahead and rewrite the POST handler for your view:

```python
user_data, error = verify_post_data(request)
if not error:
    error = verify_passwords(user_data)
if not error:
    new_user = User.create_user(user_data)
```

the nature of the `create_user` method on your User model is left as an exercise for the reader :)